tests/model_selection/scaffold_split.py

```diff
@@ -0,0 +1,215 @@
+    train, test = scaffold_train_test_split(
+        all_molecules, train_size=0.7, test_size=0.3
+    )
+    assert len(train) == 7
+    assert len(test) == 3
+
+
+def test_train_test_split_total_molecule_count(all_molecules):
+    train_split, test_split = scaffold_train_test_split(
+        all_molecules, train_size=0.8, test_size=0.2
+    )
+    assert len(train_split) + len(test_split) == len(all_molecules)
+    assert len(train_split) == 8
+    assert len(test_split) == 2
+
+
+def test_scaffold_train_test_split_with_additional_data(all_molecules, additional_data):
+    train_set, test_set, additional = scaffold_train_test_split(
+        all_molecules, *additional_data, train_size=0.7, test_size=0.3
+    )
+
+    train_additional = additional[0]
+    test_additional = additional[1]
+
+    assert len(train_set) == 7
+    assert len(test_set) == 3
+    assert len(train_additional) == 7
+    assert len(test_additional) == 3
+
+
+def test_scaffold_train_valid_test_split_with_additional_data(
+    all_molecules, additional_data
+):
+    train_set, valid_set, test_set, additional = scaffold_train_valid_test_split(
+        all_molecules, *additional_data, train_size=0.7, valid_size=0.2, test_size=0.1
+    )
+
+    train_additional = additional[0]
+    valid_additional = additional[1]
+    test_additional = additional[2]
+
+    assert len(train_set) == 7
+    assert len(valid_set) == 2
+    assert len(test_set) == 1
+    assert len(train_additional) == 7
+    assert len(valid_additional) == 2
+    assert len(test_additional) == 1
+
+
+def test_train_valid_test_split_total_molecule_count(all_molecules):
+    train_split, valid_split, test_split = scaffold_train_valid_test_split(
+        all_molecules, train_size=0.8, valid_size=0.1, test_size=0.1
+    )
+
+    assert len(train_split) + len(valid_split) + len(test_split) == len(all_molecules)
+    assert len(train_split) == 8
+    assert len(valid_split) == 1
+    assert len(test_split) == 1
+
+
+def test_test_split_smaller_than_train_split(all_molecules):
+    train_split, test_split = scaffold_train_test_split(
+        all_molecules, train_size=0.7, test_size=0.3
+    )
+
+    assert len(train_split) > len(test_split)
+    assert len(train_split) == 7
+    assert len(test_split) == 3
+
+
+def test_train_split_larger_than_valid_and_test_splits(all_molecules):
+    train_split, valid_split, test_split = scaffold_train_valid_test_split(
+        all_molecules, train_size=0.7, valid_size=0.2, test_size=0.1
+    )
+
+    assert len(train_split) > len(valid_split)
+    assert len(valid_split) > len(test_split)
+    assert len(train_split) == 7
+    assert len(valid_split) == 2
+    assert len(test_split) == 1
+
+
+def test_scaffold_creation_total_count(all_molecules):
+    scaffolds = _create_scaffolds(all_molecules)
+    assert len(scaffolds) <= len(all_molecules)
+
+
+def test_no_ring_molecules(no_ring_molecules):
+    scaffolds = _create_scaffolds(no_ring_molecules)
```

Comment:
You use those molecules only in this one place, change to `smiles_list` and paste here instead of fixture